org.eclipse.dawnsci.analysis.api/src/org/eclipse/dawnsci/analysis/api/persistence/IPersistentFile.java

```diff
@@ -128,12 +161,26 @@
 	 * Method to set the axes<br>
 	 * This will write the data to entry/data<br>
 	 * If the axes already exist, they will be overwritten.<br>
-	 * 
+	 * <p>
+	 * <b>Important:</b> the axes should be in order of the dimensions of the dataset
 	 * @param axes
 	 * @throws Exception 
 	 */
 	public void setAxes(List<? extends IDataset> axes) throws Exception;
 
+	/**
+	 * Method to set the axes<br>
+	 * This will write the data to entry/data<br>
+	 * If the axes already exist, they will be overwritten.<br>
+	 * <p>
+	 * <b>Important:</b> the axes should be in order of the dimensions of the dataset
+	 * @param axes
+	 * @throws Exception 
+	 */
+	default public void setAxes(IDataset... axes) throws Exception {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Define and throw a dedicated exception instead of using a generic one. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00112)
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Reorder the modifiers to comply with the Java Language Specification. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AModifiersOrderCheck)